src/rgw/rgw_op.cc

```diff
@@ -1404,16 +1467,31 @@ void RGWGetObj::execute()
 
   start = ofs;
 
+  if (need_decompress) {
+
+    if (cs_info.blocks.size() == 0) {
+      lderr(s->cct) << "ERROR: no info about compression blocks, cannot decompress" << dendl;
+      op_ret = -EIO;
```

Comment:
Okay, I ran s3tests again and wasn't able to reproduce the failures. Though it would be nice to check for len=0 on PutObj, and avoid writing a compression xattr for the object and bucket in that case.